src/blocks/service_status.rs

```diff
@@ -0,0 +1,143 @@
+//! Display the status of a service
+//!
+//! Right now only `systemd` is supported.
+//!
+//! # Configuration
+//!
+//! Key | Values | Default
+//! ----|--------|--------
+//! `driver` | Which init system is running the service. Available drivers are: `"systemd"` | `"systemd"`
+//! `service` | The name of the service | **Required**
+//! `format` | A string to customise the output of this block. See below for available placeholders. | `" $service $active_state "`
+//!
+//! Placeholder    | Value                     | Type   | Unit
+//! ---------------|---------------------------|--------|-----
+//! `service`      | The name of the service   | Text   | -
+//! `active_state` | The status of the service | Text   | -
```

Comment:
1. I think it's better to just have two formats: `active_format` and `inactive_format`.
2. Since `service` is just a string from the config, is it really necessary?

Reply:
I think having `active_format` and `inactive_format` is fine (or perhaps `format_active` and `format_inactive`). What would you make them default to?
I like having `" $service $active_state "` as a default for both. That way if you don't want to mess with the config to define the format if you don't want to (or  just want to set it to `""` to hide it).

In order to allow that default we'd have to use either:
* `format!(" {} $active_state ", config.service.clone()).as_str()`
* `" $service $active_state "`

I prefer just keeping `$service` as a placeholder.


Here are a few example configs that I though someone might use:

```toml
[[block]]
block = "service_status"
service = "cups"

[[block]]
block = "service_status"
service = "cups"
format_active = ""
format_inactive = " cups stopped running "
```

Reply:
I don't really see the point of `$active_state`. The defaults could be
```toml
format_active = " $service active "
format_inactive = " $service inactive  "
```
or even
```toml
format_active = " $service "
format_inactive = " $service "
```
because the state is duplicated by `widget.state`.